src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala

```diff
@@ -0,0 +1,271 @@
+import org.hammerlab.guacamole.assembly.DeBruijnGraph
+import org.hammerlab.guacamole.filters.GenotypeFilter.GenotypeFilterArguments
+import org.hammerlab.guacamole.reads.{ MappedRead, Read }
+import org.hammerlab.guacamole.reference.{ ReferenceBroadcast, ReferenceGenome }
+import org.hammerlab.guacamole.variants.{ Allele, AlleleConversions, AlleleEvidence, CalledAllele }
+import org.hammerlab.guacamole.windowing.SlidingWindow
+import org.kohsuke.args4j.{ Option => Args4jOption }
+
+import scala.collection.JavaConversions._
+
+/**
+ *
+ * Simple assembly based germline variant caller
+ *
+ */
+object GermlineAssemblyCaller {
+
+  class Arguments extends GermlineCallerArgs with GenotypeFilterArguments {
+
+    @Args4jOption(name = "--kmer-size", usage = "Length of kmer used for DeBrujin Graph assembly")
+    var kmerSize: Int = 45
+
+    @Args4jOption(name = "--snv-window-range", usage = "Number of bases before and after to check for additional matches or deletions")
+    var snvWindowRange: Int = 20
+
+    @Args4jOption(name = "--min-average-base-quality", usage = "Minimum average of base qualities in the read")
+    var minAverageBaseQuality: Int = 20
+
+    @Args4jOption(name = "--min-alignment-quality", usage = "Minimum alignment qualities of the read")
+    var minAlignmentQuality: Int = 30
+
+    @Args4jOption(name = "--reference-fasta", required = false, usage = "Local path to a reference FASTA file")
+    var referenceFastaPath: String = null
+
+    @Args4jOption(name = "--min-area-vaf", required = false, usage = "Minimum variant allele frequency to investigate area")
+    var minAreaVaf: Int = 5
+
+    @Args4jOption(name = "--min-occurrence", required = false, usage = "Minimum occurrences to include a kmer ")
+    var minOccurrence: Int = 3
+
+  }
+
+  object Caller extends SparkCommand[Arguments] {
+    override val name = "germline-assembly"
+    override val description = "call germline variants by assembling the surrounding region of reads"
+
+    /**
+     * Extract alignments that are mismatches, insertions or deletions
+     * @param alignment
+     * @return Filtered set of alignments that are mismatches, insertions or deletions
+     */
+    def getVariantAlignments(alignment: ReadAlignment): Seq[(AlignmentState, Int)] = {
+      alignment
+        .alignments
+        .zipWithIndex
+        .filter(tup => AlignmentState.isGapAlignment(tup._1) || tup._1 == AlignmentState.Mismatch)
+    }
+
+    /**
+     *
+     * @param graph An existing DeBruijn graph of the reads
+     * @param currentWindow Window of reads that overlaps the current loci
+     * @param kmerSize Length kmers in the DeBruijn graph
+     * @param minOccurrence Minimum times a kmer must appear to be in the DeBruijn graph
+     * @param expectedPloidy Expected ploidy, or expected number of valid paths through the graph
+     * @return A set of variants in the window
+     */
+    def discoverHaplotypes(graph: Option[DeBruijnGraph],
```

Comment:
This was part of a plan to have the graph updated each time rather than rebuilt each time, but this was not implemented. I can simplify this more now and remove it until it is actually implemented.